### PR TITLE
Fix deprecated GitHub action commands

### LINKED ISSUE #186: [CT-1435] [Bug] GitHub Actions: Deprecating save-state and set-output commands
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

GitHub Actions: Deprecating save-state and set-output commands, these should be replaced with the proposed solution.
See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/.

### Expected Behavior

We should not see the warnings.

### Steps To Reproduce

Run all GH checks.

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt:
```

### What Homebrew formula are you trying to install?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kevinneville**
resolves #186 

### Description

As communicated by Github in https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ these commands should be replaced.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR

### CODE REVIEW

File: .github/workflows/formula.yml [hunk 1]
```diff
@@ -88,28 +88,28 @@ jobs:
...
-          echo ::set-output name=formula-resources-file::"$RESOURCES_FILE_PATH"
+          echo "formula-resources-file=${RESOURCES_FILE_PATH}" >> $GITHUB_OUTPUT
 
       - name: Generate formula file name
         id: gen-filename
         run: |
           if [ ${{ steps.semver.outputs.is-pre-release }} == 1 ]
           then
             echo "This is a prelease"
-            echo ::set-output name=filename-with-version::"Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.base-version }}-${{ steps.semver.outputs.pre-release }}.rb"
```
Comment (kevinneville) [inline .github/workflows/formula.yml:99]: Please have an extra look at row 99, 102 & 104, I am not 100% sure about those.

File: .github/workflows/formula.yml [hunk 2]
```diff
@@ -88,28 +88,28 @@ jobs:
...
+            echo filename-with-version="Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.base-version }}-${{ steps.semver.outputs.pre-release }}.rb" >> $GITHUB_OUTPUT
```
Comment (emmyoop) [inline .github/workflows/formula.yml:99]: FILENAME_WITH_VERSION="Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.base-version }}-${{ steps.semver.outputs.pre-release }}.rb"

File: .github/workflows/formula.yml [hunk 3]
```diff
@@ -88,28 +88,28 @@ jobs:
...
           else
             echo "This is a final release"
-            echo ::set-output name=filename-with-version::"Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.version }}.rb"
+            echo filename-with-version="Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.version }}.rb" >> $GITHUB_OUTPUT
```
Comment (emmyoop) [inline .github/workflows/formula.yml:102]:
```suggestion
            FILENAME_WITH_VERSION="Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.version }}.rb"
```

File: .github/workflows/formula.yml [hunk 4]
```diff
@@ -88,28 +88,28 @@ jobs:
...
           fi
-          echo ::set-output name=filename::"Formula/${{ github.event.inputs.dbt-package }}.rb"
+          echo filename="Formula/${{ github.event.inputs.dbt-package }}.rb" >> $GITHUB_OUTPUT
```
Comment (emmyoop) [inline .github/workflows/formula.yml:104]:
```suggestion
          echo "filename-with-version=${FILENAME_WITH_VERSION}" >> $GITHUB_OUTPUT
          FILENAME="Formula/${{ github.event.inputs.dbt-package }}.rb"
          echo "filename=${FILENAME}" >> $GITHUB_OUTPUT
```

Comment (emmyoop) [reply .github/workflows/formula.yml:104]: The above three changes all go together and make the assignment a little more clear.  What you had wasn't wrong but this makes the workflow a little easier to read.

File: .github/workflows/installation-tests.yml [hunk 5]
```diff
@@ -1,11 +1,11 @@
 # **what?**
 # This workflow installs the latest version of each dbt adapter from homebrew.
-# It then run 'dbt --version' to verify the installation was successful. 
+# It then run 'dbt --version' to verify the installation was successful.
```
Comment (emmyoop) [inline .github/workflows/installation-tests.yml:3]:
```suggestion
# It then runs 'dbt --version' to verify the installation was successful.
```

**Review by emmyoop — CHANGES_REQUESTED** [review]
Thanks for updating this for the deprecation and also doing a little cleanup in formatting on the other workflow!  I had a small reorganization request that I'll go ahead and merge in to get this added to main.

File: .github/workflows/formula.yml (hunk 2 above)
Comment (emmyoop) [inline .github/workflows/formula.yml:99]:
```suggestion
            FILENAME_WITH_VERSION="Formula/${{ github.event.inputs.dbt-package }}@${{ steps.semver.outputs.base-version }}-${{ steps.semver.outputs.pre-release }}.rb"
```

**Review by emmyoop — APPROVED** [review]
@kevinneville thanks so much for helping clean up these workflows!

**Pull request merged**